core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -371,10 +371,10 @@ class HttpGoogleServicesDAO(
           // No storage logs, so make sure that the bucket is actually empty
           val fetcher = getStorage(getBucketServiceAccountCredential).objects.list(bucketName).setMaxResults(1L)
           retryWhen500orGoogleError(() => {
-            Option(executeGoogleRequest(fetcher).getItems)
-          }) flatMap {
-            case None => Future.successful(BucketUsageResponse(BigInt(0), Option(DateTime.now())))
-            case Some(_) => Future.failed(new GoogleStorageLogException("Not Available"))
+            Option(executeGoogleRequest(fetcher).getItems).exists(!_.isEmpty)
+          }) flatMap { notEmpty =>
+            if (notEmpty) Future.failed(new GoogleStorageLogException("Not Available"))
+            else Future.successful(BucketUsageResponse(BigInt(0), Option(DateTime.now())))
```

Comment:
Maybe my brain is just slow at 5pm after a long weekend, but this took me a second to reason through. Would it be simpler to have something like this?
```suggestion
            Option(executeGoogleRequest(fetcher).getItems).exists(_.isEmpty)
          }) flatMap { isEmpty =>
            if (isEmpty) Future.successful(BucketUsageResponse(BigInt(0), Option(DateTime.now())))
            else Future.failed(new GoogleStorageLogException("Not Available"))
```

Reply:
Unfortunately not as that would throw if `getItems()` returned `null`.

Reply:
if `.getItems()` was null, then `Option(executeGoogleRequest(fetcher).getItems)` would be `None` and `.exists` would return false

Reply:
Maybe I'm being thick...

Reply:
Right yeah - so `isEmpty` would be false and we'd return `Future.failed`?

Reply:
Yeah that's right. It's definitely an aesthetic change more than anything, if you think the current way you have it is more clear then I'm good with that too

Reply:
Do we want to fail if `getItems()` returns null? It didn't before so that's what's confusing me...

Reply:
Ohhh okay yeah my bad, I did not think it through all the way. Forget this suggestion 🙃

Reply:
I take your point about not being obvious. I'll change it back to use a pattern match!